docs-md/operate-and-deploy/ksql-vs-ksqldb.md

```diff
@@ -0,0 +1,63 @@
+
+Please see [KLIP-8](https://github.com/confluentinc/ksql/blob/master/design-proposals/klip-8-queryable-state-stores.md) for an extended discussion on this topic.
+
+This syntax addition is incompatible with previous versions of KSQL. That is, **a streaming query in KSQL will not be interpreted correctly as a streaming query in ksqlDB**. Streaming queries in ksqlDB require the `EMIT CHANGES` modifier, which informs ksqlDB that the caller wants to receive incremental changes to the query result as it runs perpetually.
+
+The following is a comparison between a simple streaming query in KSQL versus an equivalent streaming query in ksqlDB:
+
+```sql
+-- KSQL:
+SELECT * FROM my_stream;
+
+-- ksqlDB:
+SELECT * FROM my_stream EMIT CHANGES;
+```
+
+While streaming KSQL queries will not be interpreted as streaming queries in ksqlDB, we have taken measures to mitigate the impact of this incompatibility to the greatest extent possible. In particular, **persistent streaming queries do not currently require the `EMIT CHANGES` modifier**. All persistent queries will be implicitly interpreted as if they include an `EMIT CHANGES` modifier. This implicit behavior means that persistent queries created using KSQL will run properly, as is, in ksqlDB.
+
+!!! note
+    `EMIT CHANGES` will be required for persistent queries in the future. We recommend migrating all of your persistent streaming queries to include the EMIT CHANGES modifier.
+
+Storage
+--------
+
+ksqlDB introduces changes to the underlying materialization storage that are incompatible with the storage format of previous KSQL versions. ksqlDB will therefore not be able to run over an existing KSQL deployment.
+
+HTTP Response Format
+-----------------------
+
+In previous KSQL versions, push query responses could return multiple JSON objects that were not wrapped in an array. Thus a collection of objects returned by the server was not a valid JSON document. In ksqlDB, the HTTP response format for [push queries](/concepts/queries/push) has changed such that all responses are now valid JSON documents. Push query results returned by a ksqlDB server will be wrapped in a JSON array.
+
+Additionally, push query HTTP responses in ksqlDB will now include a query result schema as well as the ID assigned to the query. Finally, null fields are no longer included in push query result rows.
+
+Topic Name Casing
+--------------------
+
+Previous KSQL versions automatically uppercased all topics names created by the KSQL server. ksqlDB does not automatically uppercase the names of topics it creates. The casing of the named objects that ksqlDB derives topic names from (e.g. persistent queries) will be preserved.
```

Comment:
Is this true? The name of the topic defaults to the name of the stream/table (potentially with a prefix, specified as a server config) which KSQL still converts to upper-case by default (unless the stream/table name is quoted).

Reply:
@vcrfxia does this commit mean something different?

31b78a8d117e9567e085f0ba71a94ad928216fb4

Reply:
That's only for the `PRINT TOPIC` command. Previously if a user submitted the statement `PRINT TOPIC foo;` KSQL would try to print the topic `FOO` but with that change KSQL now tries to print `foo`. However, if KSQL creates a topic, e.g. via `CREATE STREAM foo AS SELECT ...` then the topic name is still `FOO`.

cc @purplefox to confirm.

Reply:
Ok, seems like this section should just be removed entirely then.